### PR TITLE
docs(tradeshield): add bots doc

### PR DESCRIPTION
**Pull request by cosmic-vagabond**
### What has Changed?

* add bots spec to tradeshield module

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

File: x/tradeshield/spec/bot.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+## Check Perpetual Order Execution Criteria Logic
+
+### Requirements
+
+  - `PerpetualOrder` object contains both `trigger_price` and `perpetual_order_type`.
+  - `market_price` can be retrieved from the oracle module.
+  - If `perpetual_order_type` is `LIMITOPEN` and `market_price` is less than or equal to `trigger_price` then trigger `executeOrders` function.
+  - If `perpetual_order_type` is `LIMITCLOSE` and `market_price` is greater than or equal to `trigger_price` then trigger `executeOrders` function.
+  - If `perpetual_order_type` is `MARKETOPEN` and regardless of both `market_price` and `trigger_price` values then trigger `executeOrders` function.
```
Comment (amityadav0) [inline x/tradeshield/spec/bot.md:22]: If we don't consider price in market open and close, can we execute it when a user creates it ?
In that case, we don't have to store them and will save a step for bot too

File: x/tradeshield/spec/bot.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  - If `perpetual_order_type` is `MARKETCLOSE` and regardless of both `market_price` and `trigger_price` values then trigger `executeOrders` function.
```
Comment (amityadav0) [inline x/tradeshield/spec/bot.md:23]: same

**Comment by fenriz07** [comment]
Are these the queries to get the **SpotOrder** and **PerpetualOrder**?

- list-pending-perpetual-order
- list-pending-spot-order

**Review by fenriz07 — APPROVED** [review]

File: x/tradeshield/spec/bot.md (hunk 1 above)
Comment (cosmic-vagabond) [reply x/tradeshield/spec/bot.md:22]: good point, yes we can execute them right away, lets add a ticket to update that and I will update the doc to reflect that change

**Pull request merged**